test/fixtures/rules/std_rule_cases/L028.yml

```diff
@@ -21,15 +37,15 @@ test_pass_single_table_consistent_references_2_subquery:
 
 test_fail_single_table_reference_when_unqualified_config:
   fail_str: SELECT my_tbl.bar FROM my_tbl
-
+  fix_str: SELECT bar FROM my_tbl
   configs:
     rules:
       L028:
         single_table_references: unqualified
 
 test_fail_single_table_reference_when_qualified_config:
-  fail_str: SELECT bar FROM my_tbl
-
+  fail_str: SELECT bar FROM my_tbl WHERE foo
+  fix_str: SELECT my_tbl.bar FROM my_tbl WHERE my_tbl.foo
```

Comment:
Ohh interesting that it affects `WHERE` clause too! Did it always do that? Do we need a simple "pass" test for that?

Reply:
There is some clever stuff about managing self references in certain dialects.
Redshift can ref a created column.

That said 
```sql
SELECT 
      a,
      b,
      a + b AS col_created_right_here,
      col_created_right_here + 1 AS sub_self_ref
  FROM tbl
  ```
  Is also valid in Redshift, Snowflake and BQ 
  So.....

Reply:
Ahhh that actually could be quite a large problem.
```sql
SELECT 
      tbl.a,
      tbl.b,
      tbl.a + tbl.b AS col_created_right_here,
      col_created_right_here + 1 AS sub_self_ref  -- noqa: disable=L028
  FROM tbl
  ```
  This noqa: is very much intentional (I normally want qualified but this is actually a self ref and I know it), will it be correctly respected ?

Reply:
I think we recently discovered that `noqa` doesn't work in the YAML files.

My original comment was we have a `fail` and `fix` test case for `WHERE` reference, but no `pass` test case. Should we have one? In theory `fail`/`fix` should be sufficient but I always like to have a `pass` test case in these instances too.

Reply:
Still wonder if we should have a simple `pass_str` test for this?

Reply:
Re: `noqa` in YAML tests. This should now work correctly for the usual YAML tests (`test/rules/yaml_test_cases_test.py::test__rule_test_case`). Until recently, `pass` tests used a weird code path that skipped parts of the linter. As of a couple weeks ago, everything should work.

Reply:
Ill add a test for this.

Reply:
Test added. for fail and pass w/ noqa.

Later I'd like to revisit this / make another rule for lateral column references - they cause huge problems.

Rule like **lateral self references must begin with __self_**

then there would be some interaction with this and other rules.